pyglet_utils/debuggable.py

```diff
@@ -0,0 +1,15 @@
+class Debuggable:
+    def __init__(self, class_name, *args):
+        self.class_name = class_name
+
+        if isinstance(args[0], str):
+            self.debug_text = args[0]
+
+    # All child classes must implement their own get_debug_output
+    # By having this raise NotImplementedError, child classes that haven't overwritten this will
+    # pass compilation, but will throw an error at runtime only
+    def get_debug_output(self):
+        if self.debug_text is not None and self.class_name is not None:
+            return self.class_name + ": " + self.debug_text
+        else:
+            raise NotImplementedError
```

Comment:
Not 100% certain what this is going for.  If `class_name` is supposed to be the name of the invoked class (such as `Debuggable` if this was created directly), then I would reformat like this:

```suggestion
class Debuggable:
    
    def __init__(self, debug_text=None):
        if debug_text is not None:
            self.debug_text = str(debug_text)
    
    @property
    def class_name(self):
        return self.__class__

    # All child classes must implement their own get_debug_output
    # By having this raise NotImplementedError, child classes that haven't overwritten this will
    # pass compilation, but will throw an error at runtime only
    def get_debug_output(self):
        if self.debug_text is not None:
            return self.class_name + ": " + self.debug_text
        else:
            raise NotImplementedError(f'{self.class_name} has not implemented get_debug_output')

```